src/entrypoint/graphql-admin-server.ts

```diff
@@ -0,0 +1,39 @@
+import dotenv from "dotenv"
+import { applyMiddleware } from "graphql-middleware"
+import { and, shield } from "graphql-shield"
+
+import { baseLogger } from "../logger"
+import { setupMongoConnection } from "../mongodb"
+import { gqlAdminSchema } from "../graphql"
+import { startApolloServer, isAuthenticated, isEditor } from "./graphql-server"
+
+dotenv.config()
+
+const graphqlLogger = baseLogger.child({ module: "graphql" })
+
+export async function startApolloServerForAdminSchema() {
+  const permissions = shield(
+    {
+      Query: {
+        allLevels: and(isAuthenticated, isEditor),
+        userDetailsByPhone: and(isAuthenticated, isEditor),
+        userDetailsByUsername: and(isAuthenticated, isEditor),
+      },
+      Mutation: {
+        merchantUpdateMapInfo: and(isAuthenticated, isEditor),
+        userUpdateLevel: and(isAuthenticated, isEditor),
+        userUpdateStatus: and(isAuthenticated, isEditor),
+      },
+    },
+    { allowExternalErrors: true },
+  )
+
+  const schema = applyMiddleware(gqlAdminSchema, permissions)
+  return startApolloServer({ schema, port: 4001 })
+}
+
+setupMongoConnection()
+  .then(async () => {
+    await startApolloServerForAdminSchema()
+  })
+  .catch((err) => graphqlLogger.error(err, "server error"))
```

Comment:
This need to be migrated to the same way used in https://github.com/GaloyMoney/galoy/pull/365 and with the use of node_env validation